shap/cext/tree_shap.h

```diff
@@ -301,7 +305,7 @@ inline void tree_saabas(tfloat *out, const TreeEnsemble &tree, const Explanation
 /**
  * This runs Tree SHAP with a per tree path conditional dependence assumption.
  */
-void dense_tree_saabas(tfloat *out_contribs, const TreeEnsemble& trees, const ExplanationDataset &data) {
+inline void dense_tree_saabas(tfloat *out_contribs, const TreeEnsemble& trees, const ExplanationDataset &data) {
```

Comment:
did you notice a perf change with the inline? just curious.

Reply:
It's not for performance reasons, rather when you link two objects that both include this file the function definitions clash. Using inline or templates means this header function gets a unique name in each object.